internal/nodes/avi_model_l4_translator.go

```diff
@@ -90,8 +91,20 @@ func (o *AviObjectGraph) ConstructAviL4VsNode(svcObj *corev1.Service, key string
 		}
 	}
 	avi_vs_meta.PortProto = portProtocols
-	// Default case.
-	avi_vs_meta.ApplicationProfile = utils.DEFAULT_L4_APP_PROFILE
+
+	if appProfileAnnotation, ok := svcObj.GetAnnotations()[lib.LBSvcAppProfileAnnotation]; ok && appProfileAnnotation != "" {
+		err := lib.CheckL4RefOnController(key, lib.AppProfile, appProfileAnnotation, avicache.SharedAVIClients())
+		if err != nil {
+			utils.AviLog.Warnf("key: %v, msg: Not processing L4 service", key)
+			return nil
+		} else {
+			appProfile = appProfileAnnotation
+		}
+		avi_vs_meta.ApplicationProfile = appProfile
```

Comment:
Shall we assign the appProfileAnnotation value directly to `avi_vs_meta.ApplicationProfile` instead?